batch_job/src/batch_job_work_queue.c

```diff
@@ -108,6 +108,10 @@ static batch_job_id_t batch_job_wq_wait (struct batch_queue * q, struct batch_jo
 		{
 			return -1;
 		}
+
+		char *transactions = string_format("%s.transactions", q->logfile);
+		work_queue_specify_transactions_log(q->data, transactions);
+		free(transactions);
```

Comment:
It might be historical. For other batch systems we ensure the file is open, so when adapting for wq, this would be the way to go. But don't know, I am making stuff up.